### DISCUSSION TITLE
Does this automatically use the latest commit on facebook/demucs master branch?

### DISCUSSION THREAD

**Discussion by nofishonfriday** [xserrat/docker-facebook-demucs#25] [Q&A] [answered]
Sorry for the maybe stupid question, I'm totally new to everything Docker related.
I've set this up as per instructions in the README.md some time ago and meanwhile noticed there are a couple of new commits on the `facebook/demucs` master branch.

I'm wondering when I run `make run track=mysong.mp3` does this automatically use the _current_ latest commit on the facebook/demucs master branch, or the commit which was latest when I initially ran it?
In other words, is this automatically synced to the `facebook/demucs` master branch, if not how do I update it?

Hope my question makes sense.
Thanks.

**Comment by xserrat** [comment, marked as answer]
There's no stupid question and you're right, there's a small change pending to merge in order to use the latest changes executing only `make run track=mysong.mp3`. 

For now, you can solve the problem by running first the following:

```bash
make build
```
and then:
```bash
make run track=mysong.mp3
```

**Reply by nofishonfriday** [reply]
Excellent, thanks a lot.

**Reply by xserrat** [reply]
You're welcome! I've just merged this PR #29 to avoid for now to need to know that we have to run the `make build` first.

**Reply by nofishonfriday** [reply]
Thanks. Just to be sure:
Does it build the docker image on _every_ `make run track=mysong.mp3` now or only when necessary (remote `facebook/demucs` branch is ahead of local branch), i.e. does it auto-check if building is necessary?
Building on every run seems not optimal to me either, because here at least it takes a good while to build (just did a `make build` as per your advice above before I saw your new post).

**Reply by xserrat** [reply]
Thanks for your patience @nofishonfriday. You're right, I've created the issue #30 based on your comment because we shouldn't run the build with the `--no-cache` flag. Good catch!
